addons/sale/models/sale_order.py

```diff
@@ -788,6 +788,12 @@ def _unlink_except_draft_or_cancel(self):
                     "You can not delete a sent quotation or a confirmed sales order."
                     " You must first cancel it."))
 
+    def write(self, vals):
+        res = super().write(vals)
+        for order in self.filtered(lambda o: o.state in ('sent', 'sale')):
+            order.message_subscribe(partner_ids=order.partner_id.ids)
```

Comment:
```suggestion
        if 'partner_id' in vals and vals.get('partner_id'):
            self.filtered(lambda o: o.state in ('sent', 'sale')).message_subscribe(partner_ids=[vals.get('partner_id')])
```
The followers should only be added when we effectively write on the order partner, and can be done in batch since the values updated will be constant for all orders in self.
This will avoid triggering this mail logic when the partner was not modified ;).

Reply:
Done :)